app/editor.js

```diff
@@ -445,7 +447,20 @@ ved.cql.renderItems = function(sel, group, indexPrefix) {
         mode: 'vega-lite',
         actions: {export: false}
       };
-      vg.embed(id, opt);
+      vg.embed(id, opt, function(err, result) {
+        if(err) {
+          console.error("[CompassQL]", err);
+          return;
+        }
+
+        // unregister signals for unused views
+        if (ved.cql.views.get(id)) {
```

Comment:
This is wrong.  This only remove views if there are new views of the same id.  
You should call destroy during `exit()`